code_generation/producers/met.py

```diff
@@ -0,0 +1,148 @@
+    input=[
+        nanoAOD.MET_covYY,
+    ],
+    output=[q.metcov11],
+    scopes=["et", "mt", "tt", "em"],
+)
+MetSumEt = Producer(
+    name="MetSumEt",
+    call="basefunctions::rename<float>({df}, {input}, {output})",
+    input=[
+        nanoAOD.MET_sumEt,
+    ],
+    output=[q.metSumEt],
+    scopes=["et", "mt", "tt", "em"],
+)
+PropagateLeptonsToMet = Producer(
+    name="PropagateLeptonsToMet",
+    call="met::propagateLeptonsToMet({df}, {input}, {output}, {propagateLeptons})",
+    input=[q.met_p4, q.p4_1_uncorrected, q.p4_2_uncorrected, q.p4_1, q.p4_2],
+    output=[q.met_p4_leptoncorrected],
+    scopes=["et", "mt", "tt", "em"],
+)
+PropagateJetsToMet = Producer(
+    name="PropagateJetsToMet",
+    call="met::propagateJetsToMet({df}, {input}, {output}, {propagateJets}, {min_jetpt_met_propagation})",
+    input=[
+        q.met_p4_leptoncorrected,
+        q.Jet_pt_corrected,
+        nanoAOD.Jet_eta,
+        nanoAOD.Jet_phi,
+        q.Jet_mass_corrected,
+        nanoAOD.Jet_pt,
+        nanoAOD.Jet_eta,
+        nanoAOD.Jet_phi,
+        nanoAOD.Jet_mass,
+    ],
+    output=[q.met_p4_jetcorrected],
+    scopes=["et", "mt", "tt", "em"],
+)
+calculateGenBosonVector = Producer(
+    name="calculateGenBosonVector",
+    call="met::calculateGenBosonVector({df}, {input}, {output})",
+    input=[
+        nanoAOD.GenParticle_pt,
+        nanoAOD.GenParticle_eta,
+        nanoAOD.GenParticle_phi,
+        nanoAOD.GenParticle_mass,
+        nanoAOD.GenParticle_pdgId,
+        nanoAOD.GenParticle_status,
+        nanoAOD.GenParticle_statusFlags,
+    ],
+    output=[q.genboson_p4],
+    scopes=["et", "mt", "tt", "em"],
+)
+ApplyRecoilCorrections = Producer(
+    name="ApplyRecoilCorrections",
+    call='met::applyRecoilCorrections({df}, {input}, {output}, "{recoil_corrections_file}", "{recoil_systematics_file}", {applyRecoilCorrections}, {apply_recoil_resolution_systematic}, {apply_recoil_response_systematic}, {recoil_systematic_shift_up}, {recoil_systematic_shift_down}, {isWJets})',
+    input=[
+        q.met_p4_jetcorrected,
+        q.genboson_p4,
+        q.Jet_pt_corrected,
+    ],
+    output=[q.met_p4_recoilcorrected],
+    scopes=["et", "mt", "tt", "em"],
+)
+MetPt = Producer(
+    name="MetPt",
+    call="quantities::pt({df}, {output}, {input})",
+    input=[q.met_p4_recoilcorrected],
+    output=[q.met],
+    scopes=["et", "mt", "tt", "em"],
+)
+MetPhi = Producer(
+    name="MetPhi",
+    call="quantities::phi({df}, {output}, {input})",
+    input=[q.met_p4_recoilcorrected],
+    output=[q.metphi],
+    scopes=["et", "mt", "tt", "em"],
+)
+MetCorrections = ProducerGroup(
+    name="MetCorrections",
+    call=None,
+    input=None,
+    output=None,
+    scopes=["et", "mt", "tt", "em"],
+    subproducers=[
+        BuildMetVector,
+        MetCov00,
+        MetCov01,
+        MetCov10,
+        MetCov11,
+        MetSumEt,
+        PropagateLeptonsToMet,
+        PropagateJetsToMet,
+        calculateGenBosonVector,
```

Comment:
Let's stick to the convention to begin producer names with capital letters

Reply:
changed